### PR TITLE
Add config option for adding additional TLS fingerprints

### PR DESCRIPTION
**Pull request by NegativeMjark**

### CODE REVIEW

File: synapse/config/tls.py [hunk 1]
```diff
@@ -73,6 +89,27 @@ def default_config(self, config_dir_path, server_name, **kwargs):
...
+        #
+        # If synapse is run behind a load balancer that handles the TLS then it
+        # will be necessary to add the fingerprints of the certificates used by
+        # the loadbalancers to this list if they are different to the one
+        # synapse is using.
+        #
+        # Homeservers are permitted to cache the list of TLS fingerprints
+        # returned in the key responses. It may be necessary to publish the
+        # fingerprints of a new certificate and wait for the caches on other
+        # servers to expire before deploying it.
```
Comment (erikjohnston) [inline synapse/config/tls.py]: This worries me a bit if we're not explicit about the cache invaliadtion timings

File: synapse/config/tls.py [hunk 2]
```diff
@@ -73,6 +89,27 @@ def default_config(self, config_dir_path, server_name, **kwargs):
...
+        tls_fingerprints: []
+        #- {"sha256": "<base64_encoded_sha256_fingerprint>"}
```
Comment (erikjohnston) [inline synapse/config/tls.py]: This is a confusing way of giving an example, I was quite confused for a minute. Maybe:

```yaml
tls_fingerprints: []
# tls_fingerprints: [{"sha256": "<base64_encoded_sha256_fingerprint>"}, ...]
```

File: synapse/rest/key/v2/local_key_resource.py [hunk 3]
```diff
@@ -49,7 +47,8 @@ class LocalKey(Resource):
                     "key": # base64 encoded NACL verification key.
                 }
             }
-            "tls_certificate": # base64 ASN.1 DER encoded X.509 tls cert.
+            "tls_fingerprints": # Fingerprints of the TLS certs this server uses.
+                - {"sha256": "..."}
```
Comment (erikjohnston) [inline synapse/rest/key/v2/local_key_resource.py]: 1. This is JSON not yaml
2. Are we actually changing the API here? Won't that affect when other servers look up keys?

Comment (NegativeMjark) [inline synapse/rest/key/v2/local_key_resource.py]: We aren't changing the API. The comment is just a tiny bit out of date.

Comment (erikjohnston) [inline synapse/rest/key/v2/local_key_resource.py]: \o/

**Comment by erikjohnston** [comment]
LGTM

**Pull request merged**